### PR TITLE
[#294] Add Chucker library

### PR DESCRIPTION
**Pull request by chornerman**
https://github.com/nimblehq/android-templates/issues/294

## What happened 👀

Add Chucker library to
- Template
- Compose sample
- XML sample

## Insight 📝

- Add Chucker library to Template, Compose sample, and XML sample
- Use Hilt to provide Chucker interceptor
- We will only apply Chucker for debug builds

## Proof Of Work 📹

**Template**

https://user-images.githubusercontent.com/13492460/198258179-1db02efa-e1d9-4912-9346-68201286aa42.mp4

**Compose sample**

https://user-images.githubusercontent.com/13492460/198258852-0834b4eb-d992-4393-8174-4847c9ecbd53.mp4

**XML sample**

https://user-images.githubusercontent.com/13492460/198258920-de6f4918-0749-41da-ba59-c1c2c333254c.mp4

### CODE REVIEW

**Comment by Tuubz** [comment]
@chornerman Nice 👍 

However, the CI is failing for some reason... 💭

**Comment by chornerman** [comment]
> @chornerman Nice 👍
> 
> However, the CI is failing for some reason... 💭

@Tuubz Look good now 🚀

File: sample-compose/app/src/main/java/co/nimblehq/sample/compose/di/modules/OkHttpClientModule.kt [hunk 1]
```diff
@@ -16,12 +19,29 @@ private const val READ_TIME_OUT = 30L
...
+        @ApplicationContext context: Context
+    ): ChuckerInterceptor {
+        val chuckerCollector = ChuckerCollector(
+            context = context,
+            showNotification = true,
+            retentionPeriod = RetentionManager.Period.ONE_HOUR
+        )
+
+        return ChuckerInterceptor.Builder(context).collector(chuckerCollector)
+            .alwaysReadResponseBody(true).build()
```
Comment (Tuubz) [inline sample-compose/app/src/main/java/co/nimblehq/sample/compose/di/modules/OkHttpClientModule.kt:45]: Sames goes for the `sample-xml` & `template` 🙏 

```suggestion
        return ChuckerInterceptor.Builder(context)
                   .collector(chuckerCollector)
                   .alwaysReadResponseBody(true)
                   .build()
```

**Review by Tuubz — APPROVED** [review]
One small comment, rest LGTM 🔥 🇹🇭 🐱

**Review by luongvo — COMMENTED** [review]
apart from @Tuubz's reviews, rest lgtm

**Review by luongvo — APPROVED** [review]

**Review by kaungkhantsoe — APPROVED** [review]

**Pull request merged**